### PR TITLE
Fix some mypy errors

### PR DESCRIPTION
**Pull request by Databean**
See the commit messages for specific errors fixed. There are still some errors in chc/cmdline.

### CODE REVIEW

File: chc/invariants/CFunVarDictionary.py [hunk 1]
```diff
@@ -65,7 +65,7 @@
     'lv':lambda x:CV.LocalVariable(*x),
     'gv':lambda x:CV.GlobalVariable(*x),
     'mv':lambda x:CV.MemoryVariable(*x),
-    'mrv':lambda x:CV.MemoryRegionVariable(*x),
```
Comment (sipma) [inline chc/invariants/CFunVarDictionary.py:68]: The MemoryRegionVariable exists as a separate type in Ocaml. I will look into it and add the appropriate type to CVar. Thanks for noticing that the type is missing here.

File: chc/api/InterfaceDictionary.py [hunk 2]
```diff
@@ -94,7 +94,6 @@
     'ha': lambda x:XP.XHeapAddress(*x),    
     'i' : lambda x:XP.XInitialized(*x),
     'ir': lambda x:XP.XInitializedRange(*x),
-    'iv': lambda x:XP.XInvalidated(*x),
```
Comment (sipma) [inline chc/api/InterfaceDictionary.py:97]: XInvalidated is used in a few library summaries to indicate that the object is no longer valid, e.g., a file descriptor after closing the file. I will add the corresponding type to XPredicate. Thanks for noticing that it is missing here.

File: chc/invariants/CFunVarDictionary.py (hunk 1 above)
Comment (Databean) [reply chc/invariants/CFunVarDictionary.py]: Reverted back to MemoryRegionVariable.

File: chc/api/InterfaceDictionary.py (hunk 2 above)
Comment (Databean) [reply chc/api/InterfaceDictionary.py]: Removed that commit.

**Comment by Databean** [comment]
Thanks for reviewing!

**Pull request merged**